routes/donations.js

```diff
@@ -14,6 +15,19 @@ router.get('/', async (req, res) => {
   }
 });
 
+router.get('/verify', async (req, res) => {
+  const { email, donationId } = req.body;
+  const donation = await db.query(`SELECT email FROM donations WHERE id = $(donationId);`, {
+    donationId,
+  });
+
+  if (donation[0].email === email) {
```

Comment:
protect against queries that return empty arrays, use either optional chaining or another bool check